### PR TITLE
Docket logic

### PR DESCRIPTION
**Pull request by cvalarida**
# Docket-Hiding Logic
An appeal object is kind of a hairy beast, so to be clear, this is where I'm pulling the information to describe the screenshots below:

**Appeal type:** `appeal.attributes.type`
**Status:** `appeal.attributes.status.type`
**Active:** `appeal.attributes.active`

I don't know if any of these appeals would happen in the wild, but to demonstrate what this PR does, I've messed around with the mock data to show the different scenarios.

**Appeal type:** original
**Status:** pending_soc
**Active:** true
image
No docket because the status is a "forbidden" type.
"What's next" still shows because the appeal is still active.

**Appeal type:** cue
**Status:** bva_decision
**Active:** true
image
No docket because of the appeal type.
"What's next" still shows because the appeal is active.
Note: It looks the same as above, but I _did_ change the mock data and test.

**Appeal type:** original
**Status:** bva_decision
**Active:** false
image
No docket and no "What's next" because active is false.

# Docket Content Logic

**Front of Docket:** false
**Advanced on Docket:** false
**Appeal type:** original
image

**Front of Docket:** true
**Advanced on Docket:** false
**Appeal type:** original
image

**Front of Docket:** false
**Advanced on Docket:** true
**Appeal type:** original
image

**Front of Docket:** false
**Advanced on Docket:** false
**Appeal type:** post_cavc_remand
image

---

**To do:**
- [ ] Get the link for the "learn more" link

### CODE REVIEW

File: src/js/claims-status/components/appeals-v2/Docket.jsx [hunk 1]
```diff
@@ -2,60 +2,76 @@ import React from 'react';
...
+import { APPEAL_TYPES } from '../../utils/appeals-v2-helpers';
 
+/**
+ * @param {Number} ahead - The number of appeals ahead of this one
+ * @param {Number} total - The total number of appeals in the docket line
+ * @param {String} form9Date - The date the form 9 was sent in (or something)
+ * @param {String} docketMonth- The month that the board is looking at (or older)
+ * @param {String} appealType - The type of appeal
+ * @param {Bool}   aod - Whether the appeal is Advanced on Docket
+ * @param {Bool}   frontOfDocket - Whether the appeal is at the front of the docket
```
Comment (markgreenburg) [inline src/js/claims-status/components/appeals-v2/Docket.jsx]: Isn't it still called `front` until you rename it when you destructure the props? 🤷‍♂️

File: src/js/claims-status/components/appeals-v2/Docket.jsx [hunk 2]
```diff
@@ -2,60 +2,76 @@ import React from 'react';
...
 }
 
 Docket.propTypes = {
   ahead: PropTypes.number.isRequired,
   total: PropTypes.number.isRequired,
-  form9Date: PropTypes.string.isRequired
+  form9Date: PropTypes.string.isRequired,
+  docketMonth: PropTypes.string,
+  appealType: PropTypes.string,
+  aos: PropTypes.bool,
```
Comment (markgreenburg) [inline src/js/claims-status/components/appeals-v2/Docket.jsx]: I'm sure this was meant to be 'aod'

File: src/js/claims-status/components/appeals-v2/Docket.jsx [hunk 3]
```diff
@@ -2,60 +2,76 @@ import React from 'react';
...
+  frontOfDocket: PropTypes.bool,
```
Comment (markgreenburg) [inline src/js/claims-status/components/appeals-v2/Docket.jsx]: front? frontOfDocket? 🤷‍♂️

File: src/js/claims-status/containers/AppealsV2StatusPage.jsx [hunk 4]
```diff
@@ -28,19 +31,31 @@ const AppealsV2StatusPage = ({ appeal }) => {
...
+    STATUS_TYPES.decisionInProgress,
+    STATUS_TYPES.bvaDevelopment
+  ];
+  const hideDocketAppealTypes = [
+    APPEAL_TYPES.reconsideration,
+    APPEAL_TYPES.cue
+  ];
+  const shouldShowDocket = appealIsActive &&
+    !hideDocketStatusTypes.includes(status.type) &&
+    !hideDocketAppealTypes.includes(appealType);
```
Comment (markgreenburg) [inline src/js/claims-status/containers/AppealsV2StatusPage.jsx]: 👍

File: src/js/claims-status/containers/AppealsV2StatusPage.jsx [hunk 5]
```diff
@@ -28,19 +31,31 @@ const AppealsV2StatusPage = ({ appeal }) => {
...
-        isClosed={appealIsClosed}/>
+        isClosed={!appealIsActive}/>
       <AlertsList alerts={alerts}/>
-      {!appealIsClosed && <WhatsNext nextEvents={nextEvents}/>}
-      {!appealIsClosed && <Docket {...docket} form9Date={form9Date}/>}
-      {appealIsClosed && <div className="closed-appeal-notice">This appeal is now closed</div>}
+      {appealIsActive && <WhatsNext nextEvents={nextEvents}/>}
+      {shouldShowDocket && <Docket {...docket} aod={aod} form9Date={form9Date} appealType={appealType}/>}
+      {!appealIsActive && <div className="closed-appeal-notice">This appeal is now closed</div>}
     </div>
```
Comment (markgreenburg) [inline src/js/claims-status/containers/AppealsV2StatusPage.jsx]: ^ much nicer!

File: src/js/claims-status/components/appeals-v2/Docket.jsx (hunk 1 above)
Comment (cvalarida) [reply src/js/claims-status/components/appeals-v2/Docket.jsx]: Oh right. Good point. :+1:
I originally just had it destructured as `frontOfDocket` without renaming from `front`, but that...was broken for obvious reasons. After I fixed it, I forgot about this. Good catch!

File: src/js/claims-status/components/appeals-v2/Docket.jsx (hunk 2 above)
Comment (cvalarida) [reply src/js/claims-status/components/appeals-v2/Docket.jsx]: :man_facepalming: Yerp.

**Review by markgreenburg — APPROVED** [review]
LGTM! Nicely done, lots of nuanced business logic.

**Pull request merged**